src/tracks/playabletrack/wavetrack/ui/SampleHandle.cpp

```diff
@@ -93,8 +99,12 @@ namespace {
    {
       // Require more than 3 pixels per sample
       const auto xx = std::max<ZoomInfo::int64>(0, viewInfo.TimeToPosition(time));
+      const auto clip = wt->GetClipAtTime(time);
+      if (!clip)
+         // Don't bother the user about that with a pop-up.
+         return true;
```

Comment:
This function is used in two places, and this change, while avoiding that pop-up when you click between clips, also will change the hit test to return a handle in the case that it did not do so before.

Does clicking and dragging in the empty space between clips make any vacuous undo/redo items?

Reply:
Aha, I tried the build.  Yes it does!

So instead the hit testing function should also be changed to return null when there is no clip.

Reply:
Actually... that's an old defect, which you might fix.

I just tried released Audacity.  When the magnification does show samples, but you click and drag outside of a clip, still you do get a bogus undo/redo item.

Reply:
Actually an old problem, and I was confused about something.  `HitTest` only happens with the * tool button.  With the pencil tool button, you can click anywhere and it has always been like that.

Reply:
`SampleHandle::HitAnywhere` is also used in WaveformView.cpp in this case.

Reply:
> Actually... that's an old defect, which you might fix.

Each time `SampleHandle::Release` is called (when letting go of the mouse with pencil tool activated), a history item is pushed. The easiest solution to this I can think of would be to create a `bool mSampleMoved`, whose value is set to true whenever a sample is modified, checked when releasing, and then set to false again. That'd be something totally new, and not completely trivial to review.